### DISCUSSION TITLE
Simple LaTeX filter before PDF is produced?

### DISCUSSION THREAD

**Discussion by MHellmund** [quarto-dev/quarto-cli#4702] [Q&A] [answered]
PDF output: Is it possible to run the LaTeX file produced by quarto through a simple (e.g., shell or python) filter before quarto runs xelatex?
It is just a  small question. 
I can  live with  [LaTeX output](https://quarto.org/docs/output-formats/pdf-basics.html#latex-output) and a `post-render` script.

**Comment by mcanouil** [comment]
Could you elaborate on your real need and what you really want to do/achieve?

**Comment by MHellmund** [comment]
Formatting of code **output** cells.  
- Pandoc wraps them in a `verbatim` environment. 
- I want to use `Verbatim` from the `fancyvrb`  package which has a lot of options, e.g. `\begin{Verbatim}[xleftmargin=2em]` for better alignment with the input code. 
- `\renewenvironment{verbatim}{\begin{Verbatim}{\end{Verbatim}}` in the header makes strange TeX errors. 
- pandoc has a `listings` option (and quarto passes it through), but it is used for both code and output and stops syntax highlighting.

Furthermore, there is a small ugliness with Julia. Example:

    ---
    format:
      pdf:
        keep-tex: true
        keep-md: true
        keep-ipynb: true

    jupyter: julia-1.8
    ---

    ```{julia}
    x = 3
    println(x)
    x = 4
    ```

The generated ipynb file shows that Jupyter/IJulia produces 2 output cells of slightly different types, since writing to `stdout` and displaying the result of the last expression are handled differently.  The .tex looks like

```
\begin{verbatim}
3
\end{verbatim}

\begin{verbatim}
4
\end{verbatim}
```
and this generates a large gap between the two lines. My script merges consecutive verbatim environments. 

I guess all this can be done with a pandoc lua filter, but for me some lines of python or shell are much faster written and tested.

**Comment by mcanouil** [comment, marked as answer]
It would probably be easier to tweak the LaTeX partials to fit your needs instead of adding new LaTeX via the `include-*` YAML keys, see <https://quarto.org/docs/journals/templates.html#latex-partials>.

**Comment by MHellmund** [comment]
This is helpful. Thank you for pointing it out!
